### ISSUE TITLE
Rails 7.0.6 label_for translations not working in index views

### ISSUE THREAD

**Issue by augustosamame** [rails/rails#48629]
### Steps to reproduce

generate a new rails project
generate a scaffold: `rails g scaffold Post title body`
run migrations
modify i18n `en.yml` file with the following content:

```
en:
  hello: "Hello world"
  activerecord:
    attributes:
      post:
        title: "Title from i18n"
        body: "Body from i18n"
```

Replace the labels in both index and show views with `<%= label(:post, :body) %>`
Or you can even do the replace in the single `posts` shared partial

```
<div id="<%= dom_id post %>">
  <p>
    <strong>Title:</strong>
    <%= post.title %>
  </p>

  <p>
    <strong><%= label(:post, :body) %>:</strong>
    <%= post.body %>
  </p>

</div>

```

### Expected behavior

The body label should show 'Body from i18n' in all views

### Actual behavior

All views show correct translated label. Index view shows the incorrect (default) `body` label

I did some testing and all views are fine, but for some reason, index view fails.

### System configuration

**Rails version 7.0.6**:

**Ruby version 3.2.2**:

**Comment by augustosamame** [comment]
Figured it out, sort of.
If I add `@post = Post.new` to index controller:

```
 def index
    @posts = Post.all
    @post = Post.new
  end
```

`label(:post, :body)` starts working

So I assume the issue is that :post does not exist for index view. However, is this still not a bug?
`label(:post, :body)` can be used to keep model attributes i18n translations DRY. And having it not available in index views is kind of weird. And the workaround is even weirder. I don't want to pollute my index controller this way just to be able to use `label(:post, :body)`

**Comment by Edouard-chin** [comment]
Dug a bit on this and you are right about the translation not picked up on the `index` page. Your assumption is also correct `label(:post, :body)`, `:post` doesn't have a corresponding instance variable so this method never get called https://github.com/rails/rails/blob/2d6f523ba5d47a03ad3e3205c24a3fa13a3b8ad1/actionview/lib/action_view/helpers/tags/translator.rb#L31-L34


That being said, I was curious whether the `label` tag was semantically valid outside a form control. According to the [W3C it is valid unless a `for` attribute maps to an un-existing labelable element](https://html.spec.whatwg.org/multipage/forms.html#the-label-element).
In Rails the `label` tag always applies a `for` attribute, so if a user uses the `label` tag it also needs to create a corresponding `input` or `textarea` field. The snippet in your case is not valid html.

To summarize, I think the translation thing is a bug, but I believe to properly fix it we should first modify the `tag` helper to be used without a labelable element.

For now, I'd suggest not using the `label` tag for your use case. Maybe you can create a small helper, sth like:

```ruby
def labelled(attribute, model)
  content_tag(:label) { model.human_attribute_name(attribute) }
end
```

**Comment by augustosamame** [comment]
Thanks for the explanation. Rails syntactic sugar sometimes makes us lazy. Too much of a good thing I guess. Will switch to helper in order to fix this without polluting my controllers. Thanks!
